aptos-move/framework/aptos-framework/sources/aggregator_v2/aggregator_v2.move

```diff
@@ -52,16 +44,9 @@ module aptos_framework::aggregator_v2 {
     /// Returns a value stored in this aggregator.
     public native fun read(aggregator: &Aggregator): u128;
 
-    public native fun snapshot(aggregator: &Aggregator): AggregatorSnapshot;
-
-    public native fun snapshot_u64(aggregator: &Aggregator): AggregatorSnapshotU64;
+    public native fun snapshot(aggregator: &Aggregator): AggregatorSnapshot<u128>;
 
-    /// Returns a value stored in this aggregator.
-    public native fun read_snapshot(aggregator: &AggregatorSnapshot): u128;
-
-    /// Returns a value stored in this aggregator.
-    public native fun read_snapshot_u64(aggregator: &AggregatorSnapshotU64): u64;
+    public native fun try_snapshot_u64(aggregator: &Aggregator): Option<AggregatorSnapshot<u64>>;
```

Comment:
nit: as_snapshot_u64?

Reply:
Would keep a try in the names of the methods that can fail

Reply:
I mean `try_ as_snapshot_u64`, but seen the comment above, `try` is great